### PR TITLE
Add tarfs implementation

### PR DESCRIPTION
**Pull request by agimenez**
Hi,

Heavily inspired by the already existing zipfs implementation, I have implemented the "tar" version. It is a read-only, in-memory representation of the contents of a tarball.

Thanks!

### CODE REVIEW

**Comment by agimenez** [comment]
For some reason this is failing in AppVeyor, but I can't reproduce the panic in my development environment. Is there something I can do to fix it?

File: tarfs/tarfs_test.go [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+	{"/", []string{"sub", "testDir1", "testFile"}},
+	{"/sub", []string{"testDir2"}},
+	{"/sub/testDir2", []string{"testFile"}},
+	{"/testDir1", []string{"testFile"}},
+}
+
+var afs *afero.Afero
+
+func TestMain(m *testing.M) {
+	tf, err := os.Open("testdata/t.tar")
```
Comment (kargakis) [inline tarfs/tarfs_test.go:49]: You probably have to fix this path to work for Windows in order to make appveyor pass.

Comment (agimenez) [reply tarfs/tarfs_test.go:49]: Hi, thanks! I think that this is OK, the panic is how I handle the "pseudo-root" entry internally, as I use a literal slash, and to be honest I didn't think on Windows systems. I set up an AppVeyor account and will debug the issue.

**Comment by agimenez** [comment]
Hi, I'm closing this PR until I have the tests passing on AppVeyor, will reopen it when it's fixed

**Pull request closed without merging**